src/Components/Browser/src/RendererRegistryEventDispatcher.cs

```diff
@@ -21,10 +21,10 @@ public static class RendererRegistryEventDispatcher
         {
             var eventArgs = ParseEventArgsJson(eventDescriptor.EventArgsType, eventArgsJson);
             var renderer = RendererRegistry.Current.Find(eventDescriptor.BrowserRendererId);
-            renderer.DispatchEvent(
+            renderer.Invoke(() => renderer.DispatchEvent(
```

Comment:
Every different implementation calls dispatch and renderrootcomponent through invoke in order to set the proper SynchronizationContext

Reply:
Closures, everywhere...

Reply:
We might not need to make this change if the sync context continued to be owned by the circuit host.

For `RemoteRenderer`, events are always dispatched by the circuit host, so that could continue to be responsible for dispatching to the correct context.

For `HtmlRenderer`, there are no events. For `WebAssemblyRenderer`, there's no sync context. So there aren't any other cases where this is needed, and it's (very slightly!) advantageous not to put `WebAssemblyRenderer` through additional indirection.

Reply:
I don't feel it's terrible to have 1 closure per event. I don't think it's going to impact the performance in any measurable way, but I'll think about alternatives. When we get threads on WebAssembly we will have to do this anyways.

Reply:
I got rid of this one